frontend/packages/core/src/Input/select.tsx

```diff
@@ -204,41 +213,63 @@ const Select = ({
   const defaultIdx = defaultOption < options.length && defaultOption > 0 ? defaultOption : 0;
   const [selectedIdx, setSelectedIdx] = React.useState(defaultIdx);
 
+  // Flattens all options and sub grouped options for easier retrieval
+  const flatOptions: SelectOption[] = options
+    .map(option => (option.group ? option.group.map(groupOption => groupOption) : option))
+    .reduce((pre: any, cur: any) => pre.concat(cur), []);
```

Comment:
should the type of `pre` and `cur` be `BaseSelectOptions`?

Reply:
Typings for that block were getting to be a pain and it was starting to look a little cluttered, so refactored it and just utilized flatten from lodash instead and added typings throughout. Good call though (pretty sure it was an any because of the typing issue and planned on returning later to fix it and forgot :) )